github/data_source_github_enterprise_team.go

```diff
@@ -0,0 +1,135 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func dataSourceGithubEnterpriseTeam() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceGithubEnterpriseTeamRead,
+
+		Schema: map[string]*schema.Schema{
+			"enterprise_slug": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The slug of the enterprise.",
+			},
+			"slug": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Computed:      true,
+				ConflictsWith: []string{"team_id"},
```

Comment:
I think if you use `ExactlyOneOf` here and in `team_id` you can rely on either one being set at all times
```suggestion
				ExactlyOneOf: []string{"team_id"},
```